pkg/build/util/generate.go

```diff
@@ -155,3 +219,18 @@ func SubstituteImageReferences(build *buildapi.Build, oldImage string, newImage
 		}
 	}
 }
+
+// SubstituteImageRepoReferences uses references to an image repository to set an actual image name
+// It also clears the ImageRepo reference from the BuildStrategy, if one was set.  The imagereference
+// field will be used explicitly.
+func SubstituteImageRepoReferences(build *buildapi.Build, imageRepo kapi.ObjectReference, newImage string) {
+	switch {
+	case build.Parameters.Strategy.Type == buildapi.STIBuildStrategyType &&
```

Comment:
I'm not in favor of this and couple others like this, but I can't think of anything better, yet :wink: